formatter.go

```diff
@@ -215,6 +237,19 @@ func (f *DefaultFormatter) buildFormatData(
 		}
 	}
 
+	var enableColor bool
+	switch f.ColorMode {
+	case ColorModeAuto:
+		fd := os.Stdout.Fd()
+		enableColor = testing.Verbose() &&
```

Comment:
Let's move this logic to `fillDescription`. We can rename it to `fillGeneral` or something like this, if you think it'd be a better name.